src/main/java/com/boot/cart/service/CartService.java

```diff
@@ -157,7 +166,15 @@ public CartDTO updateProductFromCart(String email, String productName, int quant
 
         cart.setTotal(total);
 
-        return cartEntityToDto(cartRepository.save(cart));
+        cartRepository.save(cart);
+
+        List<String> userCartEntries = cart.getEntries().stream().map(CartEntry::getProductName).collect(Collectors.toList());
+
+        String productParam = String.join(",", userCartEntries);
```

Comment:
you could have mixed those 2 lines. instead of .collect(Collectors.toList) you could use .collect(Collectors.joining(",");

Reply:
done!